backend/src/services/integrationService.ts

```diff
@@ -133,27 +136,114 @@ export default class IntegrationService {
     return IntegrationRepository.findAllByPlatform(platform, this.options)
   }
 
+  static isCodePlatform(value: string): value is CodePlatform {
+    return [
+      PlatformType.GITHUB,
+      PlatformType.GITHUB_NANGO,
+      PlatformType.GITLAB,
+      PlatformType.GIT,
+      PlatformType.GERRIT,
+    ].includes(value as PlatformType)
+  }
+
   async create(data, transaction?: any, options?: IRepositoryOptions) {
     try {
       const record = await IntegrationRepository.create(data, {
         ...(options || this.options),
         transaction,
       })
 
+      const collectionService = new CollectionService({ ...this.options, transaction })
+
+      const [insightsProject] = await collectionService.findInsightsProjectsBySegmentId(
+        record.segmentId,
+      )
+
+      if (insightsProject) {
+        await this.updateInsightsProject({
+          insightsProjectId: insightsProject.id,
+          isFirstUpdate: true,
+          platform: data.platform,
+          segmentId: record.segmentId,
+          transaction,
+        })
+      }
```

Comment:
For some reason, I don't think the `findGithubInsightsForSegment` is being triggered when isFirstUpdate is set to true. I created a github integration and it didn't update any of the properties.
And I know that there were availalbe properties because I tried to create the insights project manually which calls the endpoint to get github insights and it had results.

Probably worth debugging